python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -365,22 +365,24 @@ def __get__(self, obj, cls):
             return cls._schema[name]
 
         if self._getter_name:
-            return self._bind_method(obj, self._getter_name)()
+            returnable = self._bind_method(obj, self._getter_name)()
+            if isinstance(returnable, GangaObject):
+                returnable._setParent(self)
+            return returnable
```

Comment:
It's causing infinite loops because `Job._getMasterJob` is apparently:
```python
def _getMasterJob(self):
    parent = self._getParent()
    while parent is not None:
        if isType(parent, Job):
            break
        parent = self._getParent()
    return parent
```
which loops forever if it doesn't find a parent of the correct type which is a very silly way of doing it without proper checks in place.